pkg/plugins/shell/condition.go

```diff
@@ -0,0 +1,46 @@
+package shell
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+	"github.com/updatecli/updatecli/pkg/core/scm"
+)
+
+// Condition tests if the provided command (concatenated with the source) is executed with success
+func (s *Shell) Condition(source string) (bool, error) {
+	return s.condition(source, "")
+}
+
+// ConditionFromSCM tests if the provided command (concatenated with the source) is executed with success from the SCM root directory
+func (s *Shell) ConditionFromSCM(source string, scm scm.Scm) (bool, error) {
+	return s.condition(source, scm.GetDirectory())
+}
+
+func (s *Shell) condition(source, workingDir string) (bool, error) {
+	customCommand := s.spec.Command
+	if customCommand == "" {
+		return false, fmt.Errorf(ErrEmptyCommand)
+	}
+	if source != "" {
+		customCommand += " " + source
+	}
```

Comment:
This block feels pretty unclear, could you explain it? 🙏

Reply:
Added a comment: WDYT?